pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc

```diff
@@ -0,0 +1,125 @@
+= Reading and Writing to Request Headers and Body
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+PDK enables you to both:
+
+* <<read-and-write-headers>>
+* <<read-and-write-request-bodies>>
+
+[[read-and-write-headers]]
+== Read and Write Request Headers
+
+To access the headers in both the request and the response, transform the `RequestState` or `ResponseState` to a header state by calling the method `into_headers_state()` and awaiting it. After calling either method, access and manipulate the headers by calling the functions of the `HeadersHandler` trait.
+
+[source,Rust]
+----
+pub trait HeadersHandler {
+    fn headers(&self) -> Vec<(String, String)>;
+    fn header(&self, name: &str) -> Option<String>;
+    fn add_header(&self, name: &str, value: &str);
+    fn set_header(&self, name: &str, value: &str);
+    fn set_headers(&self, headers: Vec<(&str, &str)>);
+    fn remove_header(&self, name: &str);
+}
+----
+
+You can access headers in `on_request` or `on_response` wrapped functions by implementing the following code:
```

Comment:
Are these tutorial steps? Do they need to be numbered as such? Is this an optional step? "You can" implies that it's optional.